src/ExperimentTable.js

```diff
@@ -103,33 +137,38 @@ class ExperimentTable extends React.Component {
   }
 
   render() {
-    const { searchQuery, searchedColumnIndex, selectedSearch, selectedKingdom, checkedRows } = this.state
+    const { searchQuery, searchedColumnIndex, selectedSearch, checkedRows } = this.state
+    const { selectedDropdownFilters, experimentTableFilters } = this.state
     const { orderedColumnIndex, ascendingOrder } = this.state
     const { entriesPerPage, currentPage } = this.state
     const { host, aaData, tableHeader, enableDownload, downloadTooltip } = this.props
 
     const displayedFields = tableHeader.map(header => header.dataParam)
 
-    const dataArray = selectedSearch.trim() ?
+    const tableHeaderFilteredExperiments = this.filter(this.sort(aaData), tableHeader)
 
-      this.sort(aaData).filter(data => data &&
-        Object.keys(data).map(key => displayedFields.includes(key) ? data[key] : null)
-          .some(value => value && value.toString().toLowerCase().includes(selectedSearch))) :
-      this.filter(this.sort(aaData), tableHeader)
-        .filter(data => selectedKingdom ? data.kingdom === selectedKingdom : true)
+    const selectedSearchFilteredExperiments = _.chain(tableHeaderFilteredExperiments).filter(data =>
+      data && Object.keys(data).map(key =>
+        displayedFields.includes(key) ? data[key] : null).some(value =>
+        value && value.toString().toLowerCase().includes(selectedSearch))).value()
+
+    const dataArray = _.chain(selectedSearchFilteredExperiments).filter(data =>
+      selectedDropdownFilters.every(filter =>
+        filter ? this.containsValue(data, filter.value) : true
+      )).value()
```

Comment:
You can do `.filter(...).filter(...).value()`. You can even start with `this.filter(...)`. No need to separate this into several steps. The only thing in this sort of cases is that you should be extra careful and sensible indenting your code, so that it’s easy to see where each step starts and ends. You can add comments in between such as: `// first we filter for the selected dropdowns` or whatever.

Also `dataArray` seems a bit too generic, how about `filteredElements` or `filteredArray`?

Reply:
Make sense. Renamed and refactored.